subworkflows/local/reference_indexing_multi.nf

```diff
@@ -17,39 +17,27 @@ workflow REFERENCE_INDEXING_MULTI {
-                                                    def pileupcaller_bed       = row["pileupcaller_bedfile"]   != "" ? file(row["pileupcaller_bedfile"]     , checkIfExists: true) : ""
-                                                    def pileupcaller_snp       = row["pileupcaller_snpfile"]   != "" ? file(row["pileupcaller_snpfile"]     , checkIfExists: true) : ""
-                                                    def hapmap                 = row["hapmap_file"]            != "" ? file(row["hapmap_file"]              , checkIfExists: true) : ""
-                                                    def pmd_masked_fasta       = row["pmdtools_masked_fasta"] != "" ? file(row["pmdtools_masked_fasta"], checkIfExists: true) : ""
-                                                    def pmd_bed_for_masking    = row["pmdtools_bed_for_masking"] != "" ? file(row["pmdtools_bed_for_masking"], checkIfExists: true) : ""
-                                                    def sexdet_bed             = row["sexdeterrmine_snp_bed"]  != "" ? file(row["sexdeterrmine_snp_bed"]    , checkIfExists: true) : ""
-                                                    def bedtools_feature       = row["bedtools_feature_file"]  != "" ? file(row["bedtools_feature_file"]    , checkIfExists: true) : ""
-                                                    def genotyping_gatk_dbsnp  = row["genotyping_gatk_dbsnp"]  != "" ? file(row["genotyping_gatk_dbsnp"]    , checkIfExists: true) : ""
-                                                    [ meta, fasta, fai, dict, mapper_index, circular_target, mitochondrion, capture_bed, pileupcaller_bed, pileupcaller_snp, hapmap, pmd_masked_fasta, pmd_bed_for_masking, sexdet_bed, bedtools_feature, genotyping_gatk_dbsnp ]
-                                            }
-
+    // Import reference sheet and change empty arrays to empty strings for compatibility with single reference input
+    ch_splitreferencesheet_for_branch = Channel.fromSamplesheet("fasta_sheet")
+                                    .map{
+                                        meta, fasta, fai, dict, mapper_index, circular_target, mitochondrion, capture_bed, pileupcaller_bed, pileupcaller_snp, hapmap, pmd_masked_fasta, pmd_bed_for_masking, sexdet_bed, bedtools_feature, genotyping_gatk_dbsnp ->
+                                            meta.ploidy           = meta.ploidy == !null ? meta.ploidy : ""
+                                            fai                   = fai != [] ? fai : ""
+                                            dict                  = dict != [] ? dict : ""
+                                            mapper_index          = mapper_index != [] ? mapper_index : ""
+                                            circular_target       = circular_target != [] ? circular_target : ""
+                                            mitochondrion         = mitochondrion != [] ? mitochondrion : ""
```

Comment:
Are these files or meant to be `meta` fields?

Reply:
`circular_target`and `mitochondrion` are added as new channel objects to parallel the output of reference_indexing_single subworkflow. They are converted into meta fields just before mtnucratio.